net/net.c

```diff
@@ -196,18 +196,49 @@ M0_EXPORTED(m0_net_xprt_register);
 M0_INTERNAL void m0_net_xprt_deregister(const struct m0_net_xprt *xprt)
 {
 	int i;
-
+	int j;
 	for (i = 0; i < ARRAY_SIZE(xprts); ++i) {
 		if (xprts[i] == xprt) {
-			xprts[i] = NULL;
 			if (xprt == xprt_default)
 				xprt_default = NULL;
+			for (j = i; j < ARRAY_SIZE(xprts) - 1; ++j)
+				xprts[j] = xprts[j + 1];
 			return;
 		}
 	}
 	M0_IMPOSSIBLE("Wrong xprt.");
 }
 M0_EXPORTED(m0_net_xprt_deregister);
+
+M0_INTERNAL void m0_net_print_xprt(void)
+{
+	int i;
+	for (i = 0; i < ARRAY_SIZE(xprts); ++i) {
+		if (xprts[i] != NULL)
+			M0_LOG(M0_DEBUG, "xprt name:%s", xprts[i]->nx_name);
+	}
+}
+M0_EXPORTED(m0_net_print_xprt);
+
+M0_INTERNAL bool m0_net_check_xprt(const struct m0_net_xprt *xprt)
+{
+	struct m0_net_xprt *d_xprt = NULL;
+	bool                found = false;
+	int                 i;
+
+	for (i = 0; i < ARRAY_SIZE(xprts); ++i)
+	{
+		if (xprts[i] == xprt)
+			found = true;
+	}
+
+	d_xprt = m0_net_xprt_default_get();
+	if ((found == false) && (d_xprt == xprt))
+		xprt_default = NULL;
```

Comment:
No need for this.
As I said, this is checking function. It should not modify the "xprt_default" variable.

If this happens, it means something wrong in the deregister function. It should not use this function to correct that.

Also:
we use
```
if ((!found) && (d_xprt == xprt))
		xprt_default = NULL;
```
according to coding style.

Reply:
incorporated